src/tests/ftest/security/pool_security_acl.py

```diff
@@ -0,0 +1,421 @@
+            permission: permission to be created.
+        Return:
+            entry: daos acl entry.
+        '''
+        if PERMISSIONS == "random":
+            permission = PERMISSIONS[random.randint(0, 3)]
+        if permission == "nonexist":
+            return ""
+        if "group" in usergroup:
+            entry = "A:G:" + name + "@:" + permission
+        else:
+            entry = "A::" + name + "@:" + permission
+        return entry
+
+    def cleanup_user_group(self, num_user, num_group):
+        '''
+        Deascription:
+            Remove the daos acl user and group on host.
+        Args:
+            num_user: number of user to be cleaned.
+            num_group: number of group name to be cleaned.
+        Return:
+            none.
+        '''
+        for user in range(num_user):
+            username = "daos_ci_tester_" + str(user + 1)
+            self.add_del_user(self.test_clients, "userdel", username)
+        for group in range(num_group):
+            groupname = "daos_ci_testGrp_" + str(group + 1)
+            self.add_del_user(self.test_clients, "groupdel", groupname)
+
+    def add_del_user(self, hosts, ba_cmd, user):
+        '''
+        Deascription:
+            Add or delete the daos user and group on host by sudo command.
+        Args:
+            hosts: list of host.
+            ba_cmd: linux bash command to create user or group.
+            user: user or group name to be created or cleaned.
+        Return:
+            none.
+        '''
+        bash_cmd = os.path.join("/usr/sbin", ba_cmd)
+        homedir = ""
+        if "user" in ba_cmd:
+            homedir = "-r"
+        cmd = " ".join(("sudo", bash_cmd, homedir, user))
+        result = pcmd(hosts, cmd, False)
+
+    def pool_acl_verification(self, current_user_acl, read, write):
+        '''
+        Deascription:
+            Daos pool security verification with acl file.
+            Steps:
+                (1)Setup dmg tool for creating a pool
+                (2)Generate acl file with permissions
+                (3)Create a pool with acl
+                (4)Verify the pool create status
+                (5)Get the pool's acl list
+                (6)Verify pool read operation
+                (7)Verify pool write operation
+                (8)Cleanup user and destroy pool
+        Args:
+            current_user_acl: acl with read write access credential.
+            read: expecting read permission.
+            write: expecting write permission.
+        Return:
+            pass to continue.
+            fail to report the testlog and stop.
+        '''
+
+        # (1)Create daos_shell command
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.get_params(self)
+        servers_with_ports = [
+            "{}:{}".format(host, self.port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+        self.log.info("  (1)dmg= {}".format(dmg))
```

Comment:
(pylint-logging-format-interpolation) Use % formatting in logging functions and pass the % parameters as arguments